### DISCUSSION TITLE
MyST in sphinx reference local file problem

### DISCUSSION THREAD

**Discussion by hellozimo** [executablebooks/meta#475] [Q&A] [answered]
Use MyST in sphinx
I want to reference the local files in _static in the Markdown file, but I encountered some problems. The errors are as follows:

```
test.md:6: WARNING: 'myst' reference target not found: ../../_static/test.pdf
```

As shown above, I want to quote a local PDF file in a Markdown file `test.md`, what should I do?

thanks.

**Comment by choldgraf** [comment, marked as answer]
Ah in Sphinx the easiest way to do this is with the {download} role - try that

**Reply by hellozimo** [reply]
it works, thanks.

![image](https://user-images.githubusercontent.com/13479111/133020063-4fa57592-d6cc-4019-9e1a-fb11dd54d13d.png)

**Reply by kimrojas** [reply]
Thank you for this discussion thread. I tried the same code but it seems the download icon doesn't appear?

**Comment by parmentelat** [comment]
hi there

the selected answer above worked fine for me too
<img width="748" alt="image" src="https://user-images.githubusercontent.com/4519227/212094339-970d3ad3-6bf2-4d88-b3b6-4f8e62a462b3.png">

however, given that my input is a jupyter notebook, I'm looking for a solution that would kind of look the same under both environment, because from the notebook this magic escape does not render too well:

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/4519227/212094278-6aebe874-f214-4a75-9f72-23913edae5b6.png">

if that is relevant, my notebooks are written by jupytext in md:myst format; is there any plugin I need to enable through a setting of some kind, or ?

many thanks in advance !

**Reply by NickleDave** [reply]
@parmentelat are you using MyST-NB?
https://myst-nb.readthedocs.io/en/latest/quickstart.html

If so, do you already have `myst_nb` (and **not** `myst_parser`) in your conf.py?
```
extensions = [
    ...,
    "myst_nb"
]
```

**Reply by parmentelat** [reply]
@NickleDave 
thanks for the feedback !
the thing is, I am using jupyter-book, which I realize I did not make clear enough in the first place 
anyways, jupyter-book will IIUC run sphinx under the hood but I am not aware of the exact settings that are passed to sphinx in this process; so I don't know how to answer your question :)

**Comment by NickleDave** [comment]
Jupyter book uses MyST-NB
https://jupyterbook.org/en/stable/file-types/myst-notebooks.html?highlight=MyST-NB

However I think you are asking for Jupyter Lab to render MyST in a live notebook?  
When you say "look the same under both environment", I mean.

Sorry I missed that detail before.  

I think such a plug-in is still a work in progress. See https://github.com/executablebooks/jupyterlab-myst
My understanding is that the plug-in might ultimately need to be written in javascript, see also https://myst.tools/docs/mystjs

**Reply by parmentelat** [reply]
thanks for pointing me to jupyterlab-myst, I will definitely give that a try :)

**Reply by NickleDave** [reply]
Sure, AFAICT it's more of a repo to organize dev thoughts right now but that would be one place to watch. 

Maybe @rowanc1 can comment on future plans for JupyterLab-myst integration and what tools will be used; I thought I saw a recent blog post related to this but not finding it now

**Reply by rowanc1** [reply]
There are a few comments on the current work going on for us to align all of these projects here:
https://github.com/executablebooks/jupyterlab-myst/issues/57

As you note, the frontend of Jupyter needs this accessible in javascript, so we are planning to use the `mystjs` work that is ongoing!

The latest extension build is quite exciting (I think!), you can try it [on binder](https://mybinder.org/v2/gh/executablebooks/jupyterlab-mystjs/main?urlpath=lab) and see some of the example notebooks. The extension can now see live previews and hover-references, and many of the extensions used in JupyterBook are enabled by default -- so hopefully it is a good initial experience!

**Reply by NickleDave** [reply]
Good to know, thank you!
